source/elements/oneDAL/source/data_management/data_sources/index.rst

```diff
@@ -0,0 +1,157 @@
+.. highlight:: cpp
+.. default-domain:: cpp
+
+=======
+Data Sources
+=======
+
+Data source is a concept of an out-of-memory storage for a :capterm:`dataset`.
+It is used at the data acquisition and data preparation stages for the
+following:
+
+- To extract datasets from external sources such as databases, files, remote
+  storage.
+
+- To load datasets into the device's local memory. Data do not always fit
+  the local memory, especially when processing with accelerators. A data source
+  provides the ability to load data by batches and extracts it directly into the
+  device's local memory. Therefore, a data source enables complex data analytics
+  scenarios, such as :capterm:`online computations <Online mode>`.
+
+- To filter and normalize :capterm:`feature` values that are being extracted.
+
+- To recover missing :capterm:`feature` values.
+
+- To detect :capterm:`outliers <outlier>` and recover the abnormal data.
+
+- To transform datasets into numerical representation. Data source shall
+  automatically transform non-numeric :capterm:`categorical <categorical
+  feature>` and :capterm:`continuous <continuous feature>` data values into one
+  of the numeric :capterm:`data formats <data format>`.
+
+CSV data source
+-----------------
+Class ``csv::data_source`` is an implementation of a data source type
+for which the following is true:
+
+::
+
+::
+
+   class data_source {
+   public:
+      csv_data_source(const char *file_name);
+      csv_data_source(const std::string &file_name);
+
+      char get_delimiter() const;
+      bool get_parse_header() const;
+      std::string get_file_name() const;
+
+      csv_data_source& set_delimiter(char);
+      csv_data_source& set_parse_header(bool);
+      csv_data_source& set_file_name(const char *);
+      csv_data_source& set_file_name(const std::string &);
+   };
+
+.. namespace:: onedal::csv
+.. class:: data_source
+
+   .. function:: data_source(const char *file_name)
+
+      Creates new instance of csv data source with the given :expr:`file_name`, the
+      other attributes get default values.
+
+   .. function:: data_source(const std::string &file_name)
+
+      Creates new instance of csv data source with the given :expr:`file_name`, the
+      other attributes get default values.
+
+   .. member:: char delimiter = ','
+
+      Symbol that represents delimiter between continuous feature columns in the input file.
+
+      Getter & Setter
+         | ``char get_delimter() const``
+         | ``data_source& set_delimiter(char)``
+
+   .. member:: bool parse_header = false
+
+      Flag that indicates whether or not first line in the input file should be processed
```

Comment:
I assume that csv layout is described somewhere. It means that we can add the link to the bibliography section and somewhere here highlight that subset of csv specs is supported here

Reply:
Added more detailed description of csv in the introduction section. Also added link to more detailed description of csv format.